js_modules/dagster-ui/packages/ui-core/src/assets/insights/AssetCatalogInsightsLineChart.tsx

```diff
@@ -195,6 +207,41 @@ export const AssetCatalogInsightsLineChart = React.memo(
       }),
       [renderTooltipFn, rgbColors],
     );
+
+    const chartRef = useRef(null);
+    const onClick = (event: any) => {
+      if (chartRef.current) {
+        const chart = ChartJS.getChart(chartRef.current);
+        if (!chart) {
+          return;
+        }
+
+        const clickedElements = chart.getElementsAtEventForMode(
+          event,
+          'y',
+          {axis: 'x', intersect: false},
+          true,
+        );
+        if (clickedElements.length > 0) {
+          const index = clickedElements[0]!.index;
+
+          let timeSliceSeconds = 60 * 60; // Default to 1 hour
+          if (metrics.timestamps.length >= 2) {
+            timeSliceSeconds = metrics.timestamps[1]! - metrics.timestamps[0]!;
+          }
```

Comment:
Same here:

```
const timeA = metrics.timestamps[0];
const timeB = metrics.timestamps[1];
if (timeA && timeB) {
  ...
}
```

Also, is it possible that timeA can be either later or earlier than timeB? Would it be good to have a `Math.abs` be helpful here to ensure that it's a positive slice value? (I don't actually know, just trying to check to make sure there won't be a bug case here.)

Reply:
I don't think it's possible, the endpoint should always return timestamps in a positive order.